browser/ui/webui/settings/brave_settings_leo_assistant_handler.cc

```diff
@@ -141,4 +149,32 @@ void BraveLeoAssistantHandler::HandleResetLeoData(
   AllowJavascript();
 }
 
+void BraveLeoAssistantHandler::HandleGetModels(const base::Value::List& args) {
+  std::vector<ai_chat::mojom::ModelPtr> models(
+      ai_chat::kAllModelKeysDisplayOrder.size());
+  // Ensure we return only in intended display order
+  std::transform(ai_chat::kAllModelKeysDisplayOrder.cbegin(),
+                 ai_chat::kAllModelKeysDisplayOrder.cend(), models.begin(),
+                 [](auto& model_key) {
+                   auto model_match = ai_chat::kAllModels.find(model_key);
+                   DCHECK(model_match != ai_chat::kAllModels.end());
+                   return model_match->second.Clone();
+                 });
```

Comment:
nit: instead of duplicating this, we could create a function in models.cc, something like `GetDisplayModels()`?